kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/web/struts/RequisitionAction.java

```diff
@@ -0,0 +1,108 @@
+import org.kuali.kfs.module.purap.document.PurchaseOrderDocument;
+import org.kuali.kfs.module.purap.document.RequisitionDocument;
+import org.kuali.kfs.module.purap.document.service.PurapService;
+import org.kuali.kfs.module.purap.document.web.struts.PurchasingFormBase;
+import org.kuali.kfs.module.purap.document.web.struts.RequisitionForm;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.businessobject.Building;
+import org.kuali.kfs.sys.context.SpringContext;
+
+import edu.arizona.kfs.module.purap.PurapConstants;
+import edu.arizona.kfs.module.purap.PurapPropertyConstants;
+import edu.arizona.kfs.sys.KFSPropertyConstants;
+import edu.arizona.kfs.sys.businessobject.BuildingExtension;
+
+public class RequisitionAction extends org.kuali.kfs.module.purap.document.web.struts.RequisitionAction {
+    protected static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(RequisitionDocument.class);
+    protected ParameterEvaluatorService parameterEvaluatorService;
+    
+    @Override
+    public ActionForward refresh(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        ActionForward forward = super.refresh(mapping, form, request, response);
+        RequisitionForm rqForm = (RequisitionForm) form;
+        RequisitionDocument document = (RequisitionDocument) rqForm.getDocument();
+        document.refreshReferenceObject(PurapPropertyConstants.BUILDING_OBJ);
+        document.setOrganizationAutomaticPurchaseOrderLimit(SpringContext.getBean(PurapService.class).getApoLimit(document.getVendorContractGeneratedIdentifier(), document.getChartOfAccountsCode(), document.getOrganizationCode()));
+        setDeliveryBuildingCode(document);
+        return forward;
+    }
+    
+    @Override
+    public ActionForward useOtherDeliveryBuilding(ActionMapping mapping, ActionForm form, HttpServletRequest request, HttpServletResponse response) throws Exception {
+        super.useOtherDeliveryBuilding(mapping, form, request, response);
+        PurchasingFormBase baseForm = (PurchasingFormBase) form;
+        RequisitionDocument document = (RequisitionDocument) baseForm.getDocument();
+        document.setRouteCode("");
+        document.setAddressToVendorIndicator(false);
+        return mapping.findForward(KFSConstants.MAPPING_BASIC);
+    }
+    
+    /**
+     * Updated to detect when the delivery campus code changes and to apply the 
+     * proper default delivery type using the system parameter.
+     * 
+     * It will not be set if the value is unchanged or the prior value is blank.  
+     * The reason for this is that we don't want to override the calculations
+     * which may have been made by other code creating the requisition.
+     *
+     * Also - this mod should only take effect for B2B orders.  Non-B2B orders
+     * do not have as restricted delivery options.
+     */
+    protected void setDeliveryBuildingCode(RequisitionDocument document) {
```

Comment:
Since we are not setting document deliveryBuildingCode in this method we should call it something else. Perhaps setRouteCode or generateRouteCode.